docs/adr/0015-security-fixes-non-wmde-software.md

```diff
@@ -0,0 +1,74 @@
+- [Follow deployment instructions from here](https://wikitech.wikimedia.org/wiki/How_to_deploy_code#Creating_a_Security_Patch)
+
+### 4. Releasing
+
+- (WMDE maintained code) Merge security patch into public repository.
+- Build a new release.
+- Verify the patch resolved the security problem.
+- Publish the release.
+- Announce new release.
+
+### Components Included
+
+#### MediaWiki
+
+In the case of MediaWiki or any extension (including Wikibase) on Gerrit, [triggering a new build](../../docs/topics/pipeline.md) on the release branch where the security fix has been applied and then [publishing](../../docs/topics/publishing.md) the new release should be sufficient to include the security fixes in the new docker images and tarballs.
+
+#### ElasticSearch
+
+As for ElasticSearch we first need to acknowledge the fact that upgrading to anything beyond 7.11 would not be possible due to the new [license](https://www.elastic.co/pricing/faq/licensing) and the fact that the version we currently pin releases to is [likely not receiving security fixes](https://www.elastic.co/support/eol). If however ElasticSearch were to receive security patches we simply need to update the pinned version and publish a new release from the pipeline.
+
+#### WDQS
+
+As for WDQS we only need to identify what version contains the security fix and make sure it's been made [publicly available](https://archiva.wikimedia.org/repository/releases/org/wikidata/query/rdf/service/) for us to pin it to a new release then run the pipeline and publish the resulting docker images.
+
+### Announcing the release to community
+
+All releases, major or minor should be announced on the following mailing-list:
+
+- https://lists.wikimedia.org/mailman/listinfo/wikibaseug
+
+### Recognizing the need to release
+
+To some extent the most important part of continuously delivering security fixes is to be notified of when they are available. Since most of the code shipped in the Wikibase docker images aren't WMDE maintained we need to become aware of when they are available to be included.
+
+For MediaWiki we could add some simple test to compare which is the latest stable version we have released to the one that [currently is available](https://www.mediawiki.org/wiki/Template:MW_stable_release_number). When there is a mismatch we know there is a release available.
+
+For other software without a constant release cycle or even explanation as to why the software was updated this will be much harder to accomplish.
+
+## Decision
+
+- ANY component included in the wikibase suite (docker images, tarballs) that has a security update available can generate a new release. It is up to the team to determine it's severity and if a new release is required.
+- ALL MediaWiki maintenance releases should generate a new Wikibase release.
+- Add some kind of mechanism that regularly checks what the latest stable MediaWiki version is. If the version has changed we should do a release.
```

Comment:
I personally don't think we'd need it but I don't oppose trying it out.

Reply:
i think it could be super simple

Reply:
https://github.com/wmde/wikibase-release-pipeline/pull/175/